tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestOutputHelperQueueTestOutputIntegration.cs

```diff
@@ -58,52 +60,78 @@ internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, string
         private class XUnitLogEvent : DatadogLogEvent
         {
             private readonly string _message;
-            private readonly Context? _context;
+            private readonly Span _span;
 
             public XUnitLogEvent(string message, Span span)
             {
                 _message = message;
-                _context = span is null ? null : new Context(span.TraceId, span.SpanId, span.Context.Origin);
+                _span = span;
             }
 
             public override void Format(StringBuilder sb, LogFormatter formatter)
             {
-                formatter.FormatLog<Context?>(
-                    sb,
-                    _context,
-                    DateTime.UtcNow,
-                    _message,
-                    eventId: null,
-                    logLevel: DirectSubmissionLogLevelExtensions.Information,
-                    exception: null,
-                    (JsonTextWriter writer, in Context? state) =>
-                    {
-                        if (state.HasValue)
-                        {
-                            writer.WritePropertyName("dd.trace_id");
-                            writer.WriteValue($"{state.Value.TraceId}");
-                            writer.WritePropertyName("dd.span_id");
-                            writer.WriteValue($"{state.Value.SpanId}");
-                            writer.WritePropertyName("_dd.origin");
-                            writer.WriteValue($"{state.Value.Origin}");
-                        }
-
-                        return default;
-                    });
-            }
+                using var writer = LogFormatter.GetJsonWriter(sb);
 
-            private readonly struct Context
-            {
-                public readonly ulong TraceId;
-                public readonly ulong SpanId;
-                public readonly string Origin;
+                // Based on JsonFormatter
+                writer.WriteStartObject();
 
-                public Context(ulong traceId, ulong spanId, string origin)
+                writer.WritePropertyName("ddsource", escape: false);
+                writer.WriteValue("xunit");
```

Comment:
Is this definitely correct? The default source is `csharp`, and must stay like that for typical log submission so that the correct processing happens. I see the linked doc says it needs to be `xunit`, but just confirming...

Reply:
Yes we don't want the `csharp` processor here, that's why we have a totally different format in this case (not the serilog one)